ui/src/app/machines/views/MachineDetails/MachineCommissioning/MachineCommissioningTable/MachineCommissioningTable.tsx

```diff
@@ -0,0 +1,228 @@
+import { Link } from "react-router-dom";
+
+import TableMenu from "app/base/components/TableMenu";
+import { scriptStatus } from "app/base/enum";
+import { useTrackById } from "app/base/hooks";
+import type { TSFixMe } from "app/base/types";
+import { actions as scriptResultActions } from "app/store/scriptresult";
+import scriptResultSelectors from "app/store/scriptresult/selectors";
+import type {
+  PartialScriptResult,
+  ScriptResult,
+  ScriptResultHistory,
+} from "app/store/scriptresult/types";
+
+type Props = { scriptResults: ScriptResult[] };
+
+const renderExpandedContent = (
+  result: ScriptResult,
+  history: ScriptResultHistory,
+  hasVisibleHistory: boolean
+) => {
+  return (
+    <div>
+      {hasVisibleHistory ? (
+        <table
+          role="grid"
+          className="p-table-expanding--light p-table--machine-commissioning"
+        >
+          <tbody>
+            {history[result.id]?.map((item: PartialScriptResult) => {
+              if (item.id !== result.id) {
+                return (
+                  <tr
+                    role="row"
+                    data-test="script-result-history"
+                    key={`history-${item.id}`}
+                  >
+                    <td role="gridcell"></td>
+                    <td role="gridcell"></td>
+                    <td role="gridcell">{item.runtime}</td>
+                    <td role="gridcell">{item.updated}</td>
+                    <td role="gridcell">{item.status_name}</td>
+                    <td role="gridcell"></td>
+                  </tr>
+                );
+              } else {
+                return null;
+              }
+            })}
+          </tbody>
+        </table>
+      ) : null}
+    </div>
+  );
+};
+
+const renderActions = (
+  result: ScriptResult,
+  toggleHistory: (id: ScriptResult["id"]) => void,
+  hasHistory: boolean,
+  hasVisibleHistory: boolean
+) => {
+  const links = [];
+  if (!hasHistory) {
+    return null;
+  }
+
+  if (hasHistory) {
+    if (!hasVisibleHistory) {
+      links.push({
+        children: "View previous tests",
+        onClick: () => toggleHistory(result.id),
+        "data-test": "action-menu-show-previous",
+      });
+    } else {
+      links.push({
+        children: "Hide previous tests",
+        onClick: () => toggleHistory(result.id),
+        "data-test": "action-menu-hide-previous",
+      });
+    }
+  }
+
+  return (
+    <TableMenu
+      data-test="action-menu"
+      links={links}
+      position="right"
+      title="Take action:"
+    />
+  );
+};
+
+const MachineCommissioningTable = ({ scriptResults }: Props): JSX.Element => {
+  const dispatch = useDispatch();
+
+  const history = useSelector(scriptResultSelectors.history);
+
+  const {
+    tracked: visibleHistory,
+    toggleTracked: toggleHistory,
+  } = useTrackById();
+
+  useEffect(() => {
+    const noHistory = Object.keys(history).every((k) => !history[k].length);
```

Comment:
I think this will fix the betterer error:
``` ts
const noHistory = Object.keys(history).every((k) => !history[Number(k)].length);
```

For the sake of readability I'd probably update it to this (just personal preference):
``` ts
const noHistory = Object.keys(history).every((scriptResultId) => {
  const historicalScripts = history[Number(scriptResultId)] || [];
  return historicalScripts.length === 0;
};
```

Reply:
Always happier with more readability :+1: